File: backend_project_centropsicologico_pisw2025-main/mer/src/main/java/com/sensespsicologos/mer/models/ItemInstance.java
package com.sensespsicologos.mer.models;

import com.sensespsicologos.mer.common.models.ActiveEntity;
import jakarta.persistence.*;
import lombok.*;
import lombok.experimental.SuperBuilder;


@Entity
@Table(name = "item_instances")
@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
@SuperBuilder
public class ItemInstance extends ActiveEntity {

    @Column(name = "display_int", nullable = false, unique = true)
    private Integer displayInt;

    @Column
    private String description;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "office_id", nullable = false)
    private Office office;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "item_id", nullable = false)
    private Item item;
}

File: backend_project_centropsicologico_pisw2025-main/src/interfaces/ItemInstance.ts
import { Item } from "./Item";
import { Office } from "./Office";

export interface ItemInstance {
  id: string;
  displayInt: number;

  officeId?: string;
  office?: Office;

  itemId: string;
  item: Item;

  createdAt: Date;
  updatedAt: Date;
}
